tests/mail/tests.py

```diff
@@ -960,9 +962,11 @@ def test_attach_mime_image(self):
         self.assertEqual(image_att.get_content(), b"GIF89a...")
         self.assertIsNone(image_att.get_filename())
 
-    def test_attach_mime_image_in_constructor(self):
-        image = MIMEImage(b"\x89PNG...", "png")
-        image["Content-Disposition"] = "attachment; filename=test.png"
+    def test_attach_mime_part_in_constructor(self):
```

Comment:
Note that I tried to follow the deprecation removal instructions and the following tests failed:
- `mail.tests.MailTests.test_attach_mime_part_in_constructor`
- `mail.tests.MailTests.test_decoded_attachment_text_MIMEPart`

Do these need comments on how to be adjusted? Or is something wrong with the other instructions do you think?

Reply:
This code in EmailMessage.\_\_init\_\_() was prone to confusing interpretation. I've split the compound isinstance() to clarify:

```patch
         self.attachments = []
         if attachments:
             for attachment in attachments:
-                # RemovedInDjango70Warning: MIMEBase attachment support.
-                if isinstance(attachment, (MIMEBase, email.message.MIMEPart)):
+                if isinstance(attachment, email.message.MIMEPart):
+                    self.attach(attachment)
+                elif isinstance(attachment, MIMEBase):
+                    # RemovedInDjango70Warning: MIMEBase attachment support.
                     self.attach(attachment)
                 else:
                     self.attach(*attachment)
```

Reply:
I've opened draft PR #19577 on top of this to verify the deprecation removals.